iceoryx2/src/service/static_config/message_type_details.rs

```diff
@@ -120,3 +122,124 @@ impl MessageTypeDetails {
             && self.payload.alignment <= rhs.payload.alignment
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use iceoryx2_bb_testing::assert_that;
+
+    #[test]
+    fn test_from() {
+        struct MyPayload {
+            _a: i32,
+            _b: bool,
+            _c: i64,
+        }
+
+        let got = MessageTypeDetails::from::<i32, i64, MyPayload>(TypeVariant::FixedSize);
+        let want = MessageTypeDetails{
+            header:  TypeDetail{
+                variant: TypeVariant::FixedSize,
+                type_name: "i32".to_string(),
+                size: 4,
+                alignment: 4,
+            },
+            user_header: TypeDetail{
+                variant: TypeVariant::FixedSize,
+                type_name: "i64".to_string(),
+                size: 8,
+                alignment: 8,
+            },
+            payload: TypeDetail{
+                variant: TypeVariant::FixedSize,
+                type_name: "iceoryx2::service::static_config::message_type_details::tests::test_from::MyPayload".to_string(),
+                size: 16,
+                alignment: 8,
+            },
+        };
+        assert_that!(want, eq got);
+
+        let got = MessageTypeDetails::from::<i32, bool, String>(TypeVariant::Dynamic);
```

Comment:
Using `String` in iceoryx2 would lead to a segmentation fault. But you could use `[MyPayload]` as dynamic slice.

Reply:
understood, btw not sure you are refering to the usage of `[MyPayload]` like this:
```rust
MessageTypeDetails::from::<i32, bool, &[MyPayload]>(TypeVariant::Dynamic);
```

> Using String in iceoryx2 would lead to a segmentation fault

I think we can try our best to reject the some types such as string in `MessageTypeDetails::from`, such as panic.
Not sure we can have some refinements here, but i will have a look later. How do you think about it?

Reply:
@xieyuschen The usage would be like:

```rust
MessageTypeDetails::from::<i32, bool, [MyPayload]>(TypeVariant::Dynamic);
```

> I think we can try our best to reject the some types such as string in MessageTypeDetails::from, such as panic.
Not sure we can have some refinements here, but i will have a look later. How do you think about it?

The idea was to provide a trait `ShmSend` and a corresponding proc macro. Every type that implements `ShmSend` can be used as payload type. We provide `ShmSend` for all POD types (u64, u32, bool ...) and all shared memory compatible containers from `iceoryx2_bb_containers`. The proc macro would also add `#[repr(C)]`.

Reply:
cool, that's great, could I know the issue to track this change?

Reply:
hi @elfenpiff looks your example here doesn't work. but never mind, I can skip it in this PR and test it in the following PR. 

```
        #[repr(C)]
        struct MyPayload {
            header: i32,
            _user_header: i32,
            payload: i32,
        }
        let details = MessageTypeDetails::from::<i32, i32, [MyPayload]>(TypeVariant::Dynamic);
```

```
error[E0277]: the size for values of type `[service::static_config::message_type_details::tests::test_payload_ptr_from_header::MyPayload]` cannot be known at compilation time
   --> iceoryx2/src/service/static_config/message_type_details.rs:283:60
    |
283 |         let details = MessageTypeDetails::from::<i32, i32, [MyPayload]>(TypeVariant::Dynamic);
    |                                                            ^^^^^^^^^^^ doesn't have a size known at compile-time
```